packages/autocomplete-plugin-recent-searches/src/usecases/localStorage/search.ts

```diff
@@ -14,8 +14,8 @@ function highlight<TItem extends RecentSearchesItem>({
     _highlightResult: {
       query: {
         value: query
-          ? item.query.replace(
-              new RegExp(query, 'g'),
+          ? item.query.replaceAll(
```

Comment:
this is only supported in very new browsers (only last major of Safari for example), and no IE. I know we've gone for more modern code in other parts, but this might be a bit much, wdyt? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#Browser_compatibility

Reply:
Yep I wasn't too sure about it that's why I've pushed an [other solution](https://github.com/algolia/autocomplete.js/pull/387/commits/fedc823d99366e9ab2c04cd0b7c1c54e0ed917d0#diff-63e6bfb0532b7df484ed33b4d8716c2ec6dfb1c4b708f8699ce59964def6f2f8R8-R21)

Reply:
Using `replaceAll` is a bit edgy since it's not supported on previous browser versions. It might lead to unexpected customer behaviors in production, unable to reproduce on our development browsers.

I don't think we should use it here.

Reply:
And what about the [other solution](https://github.com/algolia/autocomplete.js/pull/387/commits/fedc823d99366e9ab2c04cd0b7c1c54e0ed917d0#diff-63e6bfb0532b7df484ed33b4d8716c2ec6dfb1c4b708f8699ce59964def6f2f8R8-R21)? I will do more tests to see if it covers all the usecases.

Reply:
Yes—that would work everywhere else. It would also be fine to inline it like this [SO post](https://stackoverflow.com/a/3561711/9940315).